windows/devices/uwb/include/windows/devices/uwb/UwbAppConfiguration.hxx

```diff
@@ -97,12 +100,40 @@ private:
     UWB_APP_CONFIG_PARAM& m_parameter;
     PropertyT& m_value;
 };
+
+/**
+ * Builder class that takes in a collection of UwbAppConfigurationParameter and outputs a UWB_SET_APP_CONFIG_PARAMS
+ */
+class UwbSetAppConfigurationParametersBuilder
+{
+public:
+    UwbSetAppConfigurationParametersBuilder(uint32_t sessionId);
+
+    /**
+     * Push a UwbAppConfigurationParameter into the collection
+     */
+    template <typename PropertyT>
+    void
+    Push(UwbAppConfigurationParameter<PropertyT> param)
+    {
+        m_params.push_back(std::move(param.m_buffer));
+        m_paramsLengthSum += param.Size();
```

Comment:
This is modifying the input `UwbAppConfigurationParameter` which might be unexpected to the caller since it will leave that object in an invalid state (no buffer, but `Size()` returns the size of the prior buffer contents). 

In the end in `Publish` all of this data is copied anyway, so might it be simpler to just maintain one large buffer and copy the data into it on each `Push`? For example, instead of `std::vector<std::unique_ptr<uint8_t[]>> m_params;` we instead use `std::vector<uint8_t> m_params;` and then `Push` gets implemented like this:

```suggestion
        m_params.insert(std::end(m_params), param.Buffer(), param.Buffer() + param.Size());
```

This way you don't need the `m_paramsLengthSum` either, since `m_params.size()` tracks that.

Reply:
Oh that's a good idea